RealmTasks Apple/RealmTasks Shared/RealmColor.swift

```diff
@@ -54,13 +54,13 @@ extension Color {
     }
 }
 
-extension CollectionType where Generator.Element == Color, Index == Int {
+extension Collection where Iterator.Element == Color, Index == Int {
     func gradientColorAtFraction(fraction: Double) -> Color {
         // Ensure offset is normalized to 1
-        let normalizedOffset = max(min(fraction, 1), 0)
+        let normalizedOffset = Swift.max(Swift.min(fraction, 1.0), 0.0)
 
         // Work out the 'size' that each color stop spans
-        let colorStopRange = 1 / Double(count - 1)
+        let colorStopRange = 1.0 / Double(self.count.toIntMax() - 1)
```

Comment:
It was simply `Double(self.count)` in Swift 2, but since `self.count` was changed to type `IndexDistance` in Swift 3, that wouldn't compile.

I looked into it further and it seems `self.endIndex` is an appropriate replacement now.